specification/botservice/resource-manager/Microsoft.BotService/stable/2017-12-01/botservice.json

```diff
@@ -457,177 +795,781 @@
+        },
+        "luisAppIds": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Collection of LUIS App Ids"
+        },
+        "luisKey": {
+          "type": "string",
+          "description": "The LUIS Key"
+        }
+      },
+      "description": "The parameters to provide for the Bot.",
+      "required": [
+        "displayName",
+        "endpoint",
+        "msaAppId"
+      ]
+    },
+    "BotResponseList": {
+      "properties": {
+        "nextLink": {
+          "description": "The link used to get the next page of bot service resources.",
+          "type": "string"
+        },
+        "value": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Bot"
+          },
+          "description": "Gets the list of bot service results and their properties."
+        }
+      },
+      "description": "The list of  bot service operation response."
+    },
+    "BotChannel": {
+      "type": "object",
+      "description": "Bot channel resource definition",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/Channel",
+          "description": "The set of properties specific to bot channel resource"
+        }
+      }
+    },
+    "Channel": {
+      "type": "object",
+      "description": "Channel definition",
+      "discriminator": "channelName",
+      "properties": {
+        "channelName": {
+          "type": "string",
+          "description": "The channel name"
+        }
+      },
+      "required": [
+        "channelName"
+      ]
+    },
+    "FacebookChannel": {
+      "type": "object",
+      "description": "Facebook channel definition",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Channel"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/FacebookChannelProperties",
+          "description": "The set of properties specific to bot facebook channel"
+        }
+      }
+    },
+    "FacebookChannelProperties": {
+      "properties": {
+        "verifyToken": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Verify token"
+        },
+        "pageId": {
+          "type": "string",
+          "description": "Page id"
+        },
+        "appId": {
+          "type": "string",
+          "description": "Facebook application id"
+        },
+        "appSecret": {
+          "type": "string",
+          "description": "Facebook application secret"
+        },
+        "accessToken": {
+          "type": "string",
+          "description": "Facebook application access token"
+        },
+        "callbackUrl": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Callback Url"
+        },
+        "isEnabled": {

```

Comment:
bools are typically not recommended. Not self descriptive and dont allow for future expansion of values that can be supported. Consider - channelState or channelStatus - Enabled|Disabled and in future it can potentially support more values.